internal/pkg/views/notifications.go

```diff
@@ -0,0 +1,175 @@
+package views
+
+import (
+	"context"
+	"fmt"
+	"github.com/lawrencegripper/azbrowse/internal/pkg/style"
+	"github.com/lawrencegripper/azbrowse/pkg/armclient"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/jroimartin/gocui"
+	"github.com/lawrencegripper/azbrowse/internal/pkg/eventing"
+)
+
+func init() {
+	pendingDeletes = []pendingDelete{}
+}
+
+type pendingDelete struct {
+	display string
+	url     string
+}
+
+var pendingDeletes []pendingDelete
+var deleteMutex sync.Mutex // ensure delete occurs only once
+var gui *gocui.Gui
+
+// AddPendingDelete queues deletes for
+// delete once confirmed
+func AddPendingDelete(display, url string) {
+	if url == "" {
+		eventing.SendStatusEvent(eventing.StatusEvent{
+			Failure: true,
+			Message: "Item `" + display + "` doesn't support delete",
+			Timeout: time.Second * 5,
+		})
+		return
+	}
+	gui.Update(func(g *gocui.Gui) error {
+		deleteMutex.Lock()
+		defer deleteMutex.Unlock()
+
+		for _, i := range pendingDeletes {
+			if i.url == url {
+				eventing.SendStatusEvent(eventing.StatusEvent{
+					Failure: true,
+					Message: "Item already `" + display + "` in pending delete list",
+					Timeout: time.Second * 5,
+				})
+				return nil
+			}
+		}
+
+		pendingDeletes = append(pendingDeletes, pendingDelete{
+			url:     url,
+			display: display,
+		})
+
+		return nil
+	})
+}
+
+// ConfirmDelete delete all queued/pending deletes
+func ConfirmDelete() {
+	gui.Update(func(g *gocui.Gui) error {
+		deleteMutex.Lock()
+		defer deleteMutex.Unlock()
+
+		pending := pendingDeletes
```

Comment:
In my testing I did not receive this notification and it simply forze for ~30 seconds and then said deletion complete.

Reply:
Ah nice great spot - looks like I've messed up and somehow the requests are happening on the UI thread (need to dig into GOCUI a bit more and work out how it works). 

I've fixed this up by moving out of `Update` call and moving to a go routine then relying on the `SendStatusEvent` to trigger updates on the UI which seems to work nicely.